org.eclipse.scanning.example/src/org/eclipse/scanning/example/scannable/NexusSlit.java

```diff
@@ -0,0 +1,254 @@
+import org.eclipse.scanning.api.points.IPosition;
+import org.eclipse.scanning.api.scan.ScanningException;
+import org.eclipse.scanning.api.scan.rank.IScanRankService;
+import org.eclipse.scanning.api.scan.rank.IScanSlice;
+import org.eclipse.scanning.example.Services;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class provides an implementation which will write an NXslit to NeXus based on information from individual scannables
+ */
+public class NexusSlit extends AbstractScannable<DeviceValueMultiPosition> implements INexusDevice<NXslit> {
+
+	private static final Logger logger = LoggerFactory.getLogger(NexusSlit.class);
+
+	private String x_gap_deviceName;
+	private String y_gap_deviceName;
+
+	private IScannable<?> x_gap;
+	private IScannable<?> y_gap;
+
+	public NexusSlit() {
+		super(Services.getScannableDeviceService());
+	}
+
+	// implements IScannable<DeviceValueMultiPosition>
+
+	@Override
+	public DeviceValueMultiPosition getPosition() throws Exception {
+		DeviceValueMultiPosition position = new DeviceValueMultiPosition();
+		position.put(NXslit.NX_X_GAP, (double)x_gap.getPosition());
+		position.put(NXslit.NX_Y_GAP, (double)y_gap.getPosition());
+		return position;
+	}
+
+	@Override
+	public void setPosition(DeviceValueMultiPosition value, IPosition position) throws Exception {
+		logger.debug("setPosition({}, {}) called on {}", value, position, getName());
+
+		if (value!=null) {
+			logger.warn("non null setPosition() not expected on {}, ignoring...", getName());
+		}
+
+		if (position!=null) {
+			write(value, getPosition(), position);
+		}
+	}
+
+	private void write(DeviceValueMultiPosition demand, DeviceValueMultiPosition actual, IPosition loc) throws Exception {
+
+		// There may be actual values when there aren't demand values, but there should never be demand values when
+		// there are no actual values.
+		if (xLzValue==null || yLzValue==null) {
+			return;
+		}
+		if (actual!=null) {
+			// write actual position
+			final Dataset newXPositionData = DatasetFactory.createFromObject(actual.get(NXslit.NX_X_GAP));
+			final Dataset newYPositionData = DatasetFactory.createFromObject(actual.get(NXslit.NX_Y_GAP));
+			IScanSlice rslice = IScanRankService.getScanRankService().createScanSlice(loc);
+			SliceND xSliceND = new SliceND(xLzValue.getShape(), xLzValue.getMaxShape(), rslice.getStart(), rslice.getStop(), rslice.getStep());
+			SliceND ySliceND = new SliceND(yLzValue.getShape(), yLzValue.getMaxShape(), rslice.getStart(), rslice.getStop(), rslice.getStep());
+			if (isWritingOn()) {
+				xLzValue.setSlice(null, newXPositionData, xSliceND);
+				yLzValue.setSlice(null, newYPositionData, ySliceND);
+			}
+		}
+
+		if (xLzSet==null || yLzSet==null) {
+			return;
+		}
+		if (demand!=null) {
+			int index = loc.getIndex(getName());
+			if (index<0) {
+				throw new Exception("Incorrect data index for scan for value of '"+getName()+"'. The index is "+index);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS00112)